File: game_server/ServerProtocol/src/GC2GS/p032_GuildOp/GC2GS_032_009_ReqChgGuildName.java
package GC2GS.p032_GuildOp;

import java.nio.ByteBuffer;
/*********
 * 请求变更联盟名称
 **/
public class GC2GS_032_009_ReqChgGuildName implements ALBasicProtocolPack._IALProtocolStructure {
private String name;
private String simpleName;


public GC2GS_032_009_ReqChgGuildName() {
	name = "";
	simpleName = "";
}

public GC2GS_032_009_ReqChgGuildName(
	 String _name
	, String _simpleName
) {	name = _name;
	simpleName = _simpleName;
}

public final byte getMainOrder() { return (byte)32; }

public final byte getSubOrder() { return (byte)9; }

public String getName() { return name; }
public void setName(String _name) { name = _name; }
public String getSimpleName() { return simpleName; }
public void setSimpleName(String _simpleName) { simpleName = _simpleName; }


public final int GetBufSize() {
	int _size = 0;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(name);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(simpleName);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 2;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(name);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(simpleName);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) name = ALBasicProtocolPack.ALProtocolCommon.GetStringFromBuf(_buf);
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) simpleName = ALBasicProtocolPack.ALProtocolCommon.GetStringFromBuf(_buf);
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	ALBasicProtocolPack.ALProtocolCommon.PutStringIntoBuf(_buf, name);
	ALBasicProtocolPack.ALProtocolCommon.PutStringIntoBuf(_buf, simpleName);
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)32);
	_buf.put((byte)9);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)32);
	_recBuf.put((byte)9);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

File: game_client/Assets/Scripts/ClientProtocol/GC2GS/p032_GuildOp/GC2GS_032_009_ReqChgGuildName.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GC2GS.p032_GuildOp
{

/// <summary>
/// 请求变更联盟名称
/// </summary>
public class GC2GS_032_009_ReqChgGuildName : ALBasicProtocolPack._IALProtocolStructure {
private string name;
private string simpleName;


public GC2GS_032_009_ReqChgGuildName() {
	name = "";
	simpleName = "";
}

public GC2GS_032_009_ReqChgGuildName(
	string _name
	, string _simpleName
) {	name = _name;
	simpleName = _simpleName;
}

public byte getMainOrder() { return (byte)32; }

public byte getSubOrder() { return (byte)9; }

public string getName() { return name; }
public void setName(string _name) { name = _name; }
public string getSimpleName() { return simpleName; }
public void setSimpleName(string _simpleName) { simpleName = _simpleName; }


public int GetBufSize() {
	int _size = 0;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(name);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(simpleName);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 2;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(name);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(simpleName);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	name = _buf.getString();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	simpleName = _buf.getString();
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putString(name);
	_buf.putString(simpleName);
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)32);
	_buf.put((byte)9);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)32);
	_recBuf.put((byte)9);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("name").Append(":").Append(name.ToString()).Append(", ");
	builder.Append("simpleName").Append(":").Append(simpleName.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

